arroyo-api/src/main.rs

```diff
@@ -603,17 +570,39 @@ impl ApiGrpc for ApiServer {
         request: Request<CreatePipelineReq>,
     ) -> Result<Response<CreateJobResp>, Status> {
         let (request, auth) = self.authenticate(request).await?;
-        self.start_or_preview(request.into_inner(), false, auth)
+
+        let mut client = self.client().await?;
```

Comment:
> Excited to see this contribution and for the feature!
> 
> I have a couple of high-level thoughts about this.
> 
> The first is, for the feature as it exists in this PR, I wonder if the existing preview APIs would be sufficient to build this as a purely (or mostly) front-end feature. 

I don't believe that the preview function is going to accomplish what we're looking for.  Here are some reasons why:

Preview generates a pipeline with a TTL limit.  I would like something that doesn't have that bound on it.  The primary purpose of the source passthrough (raw) pipeline is to avoid compile times.  If this pipeline doesn't live indefinitely, bounded to the lifetime of the source, then you may have to compile this every once in a while.  I would rather have the overhead of a permanent pipeline with the trade off being that you don't *ever* need to worry about compiling this.

My biggest gripe so far is the compile times are really long for all queries, and this interrupts my flow and my ability to prototype and try new things out quickly.  Shorter iteration times lead to better solutions, at least from my experience and my style of data processing and writing these types of pipelines.

By having this as a first-class pipeline, compiled and ready to go, it lets me immediately see the output of a source.

> The preview API lets you send it a query, which it will run for a certain amount of time. You could spin up a preview pipeline with `select * from source` whenever the user clicks the preview button, which would avoid a lot of the need to track the pipeline / restart it / refresh the outputs, etc. And since it uses existing functionality it would reduce the complexity of this feature significantly.

Again, the compile times here are long to the point that it would (in my opinion) defeat the purpose of having this feature in the first place.  I want to be able to click the preview button and get immediate results, rather than having to wait for *anything* to compile, at *any* point other than when generating the pipeline and compiling when the source is created.

> The second thought is that having a source preview feature would be especially useful in the source creation flow, when a user needs to create a schema and validate it against the data.
> 
> However, currently you need a source (with a schema) in order to create a pipeline so this approach won't work.
> 
> I see a couple of paths to address this:
> 
> * Rely on a separate mechanism to tail sources that doesn't go through the whole pipeline machinery (you can see a potential start of this approach in the source testing machinery in https://github.com/ArroyoSystems/arroyo/blob/master/arroyo-api/src/testers.rs). The main downside here that it requires duplicating some of the source logic, unless it's possible to reuse it in some way.

I thought about this actually!  I thought the fastest way forward would be to implement something utilizing existing structs and code like we have for first-class pipelines, but I agree: logically it makes the most sense to me to just "tail" a "source", meaning we just visualize *actual raw stream output* instead of generating an entire pipeline for the same.

> * @jacksonrnewhouse is going to be working on adding support for raw schemas for sources, with `JSONB` functions to query them. Once that's available, we might be able to rely on that to tail the data.

Looking forward to seeing this.

> At a higher level, for these initial PRs it will be easier to review if the first pass is a pretty direct implementation of the feature without a lot of refactoring.

Totally.  I got carried away and should have separated it out.  Refactor should exist in a separate PR.  I'll try to go through and pare this down to just the essential changes.

> Happy to jump on a call as well to talk through this stuff if that would be helpful.

Let's find some time to chat about it!  Would love to introduce myself.

---

For now, I'm going to continue down this path, until we get some other logic for tailing streams without having an official "source". I do strongly believe that this is the fastest, most efficient way to get this feature into the product and due to the nature of the change, it could be refactored and these raw pipelines are decoupled enough from the rest of the system that they could be removed and revamped in favor of a "tail" output for a "stream" (not an official "source").

Let's